Doc/reference/datamodel.rst

```diff
@@ -441,7 +441,7 @@ Sets
 
    These represent a mutable set. They are created by the built-in :func:`set`
    constructor and can be modified afterwards by several methods, such as
-   :meth:`~set.add`.
+   :meth:`~frozenset.add`.
```

Comment:
@serhiy-storchaka objected when I tried this before ;) see https://github.com/python/cpython/pull/112737#pullrequestreview-1764409629

Reply:
Not sure I correctly understand the reason.

Reply:
Yeah, I've seen a docs user confused about set and frozenset being combined, and that the anchor links only work for one of them.

We should definitely come up with a fix... (In another PR)

Reply:
Apart from anything else, it looks extremely strange to people reading through the source for the docs.

The Sphinx reference warning here is also indicating a deeper problem here: it's impossible to properly link to the docs for `set.add` in the way people would expect. That's not just a problem for us in the CPython repo; it also affects users of intersphinx, who'll be linking to cpython docs from other repos. Rather than just covering the problem up by changing the reference to "frozenset.add" (which doesn't exist, as you know ;), we should fix the markup in `stdtypes.rst` so that it's possible to directly link to `set.add`

Reply:
> Not sure I correctly understand the reason.

Because `frozenset` does not have the `add()` method.

Reply:
>  looks extremely strange to people reading through **the source** for the docs.
> Because frozenset does not have the add() method.

ok

> We should definitely come up with a fix... (In another PR)

Maybe there is some sphinx hack, but so far I see only one solution: split the class directive for set and frozenset.  In this way we don't have such problem for list/tuple pair.

Reply:
If you'd prefer to postpone fixing this issue for now, one option is to add `set.add` to a different, more targeted allowlist: https://github.com/python/cpython/blob/05e47202a34e6ae05e699af1083455f5b8b59496/Doc/conf.py#L175-L254

Reply:
> Maybe this worth an issue, however.

Agreed!